### DISCUSSION TITLE
[QA] React Server Components

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25694] [Announcement]
#### Feature owner: @shilman 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by ethriel3695** [comment]
Project: https://github.com/vercel/next-react-server-components
Version: 8.0.0-alpha.14
Bug/Feedback: The project works as expected. Client components load in Storybook as expected. Server components load as well. However, server components break the docs when there is more than one story, as expected. I grabbed a random RSC component from a tutorial on the web from Josh Comeau and added it to my project. I then added the story for that component, which works as expected.

**Server Component with broken Docs page**
<img width="1511" alt="Screenshot 2024-01-27 at 8 26 46 AM" src="https://github.com/storybookjs/storybook/assets/12070791/3ef537d2-b4f2-47f2-997e-734bde5896cd">

**Client Comment Component with a working Docs page and stories**
<img width="1511" alt="Screenshot 2024-01-27 at 8 37 56 AM" src="https://github.com/storybookjs/storybook/assets/12070791/50517cc8-e5c9-4640-b7fa-aecd769f2a04">
<img width="1511" alt="Screenshot 2024-01-27 at 8 37 51 AM" src="https://github.com/storybookjs/storybook/assets/12070791/7ebc3534-6b70-47de-8377-59a5569cd735">


**Server Footer Component but not async with a working Docs page**
<img width="1511" alt="Screenshot 2024-01-27 at 8 37 40 AM" src="https://github.com/storybookjs/storybook/assets/12070791/d91f9a49-7f5f-460c-9ae9-adacada9b068">

**Comment by ethriel3695** [comment]
For anyone who finds this Discussion.
@shilman  let me know that we found the issue.

The docs issue is a limitation of React.

The workaround is here https://github.com/storybookjs/storybook/issues/25594#issuecomment-1913903578

[Comment on #25594 React: RSCs don't render correctly in Docs](https://github.com/storybookjs/storybook/issues/25594#issuecomment-1913903578)

This is a limitation of React, which cannot render multiple async component roots on the client.

Repro: https://github.com/shilman/vite-rsc-test

To work around this, you can disable inline docs rendering:

`parameters: { docs: { story: { inline: false } } }`

I tested it out on my RSC components, and the workaround is successful.

Here is a screenshot with docs issues and console errors.
<img width="1511" alt="Screenshot 2024-01-29 at 8 20 41 AM" src="https://github.com/storybookjs/storybook/assets/12070791/27745b4d-c534-45ff-a49b-96fbfc5287bf">

Here is an after screenshot, with the docs rendering as expected.
<img width="1511" alt="Screenshot 2024-01-29 at 8 21 07 AM" src="https://github.com/storybookjs/storybook/assets/12070791/3ccbd347-9865-4e01-835b-7680e92167f2">
